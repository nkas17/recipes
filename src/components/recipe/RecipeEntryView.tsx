/* eslint-disable react/require-default-props */
import React from 'react';
import TextInput from '../common/TextInput';
import TextArea from '../common/TextArea';
import SelectInput from '../common/SelectInput';
import { Recipe } from '../../types/recipe';
import { Category } from '../../types/category';

interface RecipeEntryViewProps {
  recipe: Recipe;
  categories: Category[];
  onSave: () => void;
  onChange: () => void;
  onCancel: () => void;
  saving?: boolean;
  errors?: {
    title: string;
    description: string;
    ingredients: string;
    directions: string;
    category: string;
  };
}

function RecipeEntryView({
  recipe,
  categories,
  onSave,
  onChange,
  onCancel,
  saving = false,
  errors = { title: '', description: '', ingredients: '', directions: '', category: '' },
}: RecipeEntryViewProps) {
  return (
    <form>
      <TextInput
        name="title"
        label="Title"
        value={recipe.title}
        onChange={onChange}
        error={errors.title}
      />
      <TextInput
        name="description"
        label="Description"
        value={recipe.description}
        onChange={onChange}
        error={errors.description}
      />
      <TextArea
        name="ingredients"
        label="Ingredient List"
        rows="10"
        value={recipe.ingredients}
        onChange={onChange}
        error={errors.ingredients}
      />
      <TextArea
        name="directions"
        label="Directions"
        rows="10"
        value={recipe.directions}
        onChange={onChange}
        error={errors.directions}
      />
      <SelectInput
        name="category"
        label="Category"
        value={recipe.category}
        defaultOption="select category"
        options={categories}
        onChange={onChange}
        error={errors.category}
      />
      <button
        id="save"
        type="submit"
        disabled={saving}
        className="btn btn-primary"
        onClick={onSave}
      >
        {saving ? 'saving...' : 'save'}
      </button>
      <button
        id="cancel"
        type="button"
        disabled={saving}
        className="btn btn-link"
        onClick={onCancel}
      >
        cancel
      </button>
    </form>
  );
}

export default RecipeEntryView;
