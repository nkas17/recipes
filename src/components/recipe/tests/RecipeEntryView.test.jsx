/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import toJSON from 'enzyme-to-json';
import { shallow } from 'enzyme';
import RecipeEntryView from '../RecipeEntryView';

const defaultProps = {
  recipe: {},
  saving: false,
  errors: {},
  categories: [],
  onSave: () => {},
  onChange: () => {},
  onCancel: () => {},
};

describe('RecipeEntryView', () => {
  it('renders correctly', () => {
    const props = {
      ...defaultProps,
    };
    const tree = shallow(<RecipeEntryView {...props} />);
    expect(toJSON(tree)).toMatchSnapshot();
  });

  it('renders correctly when saving', () => {
    const props = {
      ...defaultProps,
      saving: true,
    };
    const tree = shallow(<RecipeEntryView {...props} />);
    expect(toJSON(tree)).toMatchSnapshot();
  });
});
