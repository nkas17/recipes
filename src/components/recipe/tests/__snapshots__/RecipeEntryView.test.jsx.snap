// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeEntryView renders correctly 1`] = `
<form>
  <TextInput
    error=""
    label="Title"
    name="title"
    onChange={[Function]}
    placeholder=""
    value=""
  />
  <TextInput
    error=""
    label="Description"
    name="description"
    onChange={[Function]}
    placeholder=""
    value=""
  />
  <TextArea
    cols=""
    error=""
    label="Ingredient List"
    name="ingredients"
    onChange={[Function]}
    placeholder=""
    readonly={false}
    rows="10"
    value=""
  />
  <TextArea
    cols=""
    error=""
    label="Directions"
    name="directions"
    onChange={[Function]}
    placeholder=""
    readonly={false}
    rows="10"
    value=""
  />
  <SelectInput
    defaultOption="select category"
    error=""
    label="Category"
    name="category"
    onChange={[Function]}
    options={Array []}
    value=""
  />
  <button
    className="btn btn-primary"
    disabled={false}
    id="save"
    onClick={[Function]}
    type="submit"
  >
    save
  </button>
  <button
    className="btn btn-link"
    disabled={false}
    id="cancel"
    onClick={[Function]}
    type="button"
  >
    cancel
  </button>
</form>
`;

exports[`RecipeEntryView renders correctly when saving 1`] = `
<form>
  <TextInput
    error=""
    label="Title"
    name="title"
    onChange={[Function]}
    placeholder=""
    value=""
  />
  <TextInput
    error=""
    label="Description"
    name="description"
    onChange={[Function]}
    placeholder=""
    value=""
  />
  <TextArea
    cols=""
    error=""
    label="Ingredient List"
    name="ingredients"
    onChange={[Function]}
    placeholder=""
    readonly={false}
    rows="10"
    value=""
  />
  <TextArea
    cols=""
    error=""
    label="Directions"
    name="directions"
    onChange={[Function]}
    placeholder=""
    readonly={false}
    rows="10"
    value=""
  />
  <SelectInput
    defaultOption="select category"
    error=""
    label="Category"
    name="category"
    onChange={[Function]}
    options={Array []}
    value=""
  />
  <button
    className="btn btn-primary"
    disabled={true}
    id="save"
    onClick={[Function]}
    type="submit"
  >
    saving...
  </button>
  <button
    className="btn btn-link"
    disabled={true}
    id="cancel"
    onClick={[Function]}
    type="button"
  >
    cancel
  </button>
</form>
`;
